vezba-stabla/src/StabloProgram.java

```diff
@@ -0,0 +1,203 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.svetovid.Svetovid;
+
+class Stablo {
+    private static class Cvor {
+    	Automobil info;
+    	Cvor levo;
+    	Cvor desno;
+    	
+    	public Cvor(Automobil info, Cvor levo, Cvor desno) {
+    		this.info = info;
+    		this.levo = levo;
+    		this.desno = desno;
+    	}
+    	
+    	public Cvor kopija() {
+    		return new Cvor(info, levo, desno);
+    	}
+    }
+
+    private Cvor koren;
+
+    // konstruktor za prazno stablo
+    public Stablo() {
+    	this.koren = null;
+    }
+    
+    public Stablo(Automobil auto) {
+    	this.koren = new Cvor(auto, null, null);
+    }
+    
+    protected Stablo(Cvor cvor) {
+    	this.koren = cvor;
+    }
+    
+    
+    // ------ dodati opisane metode ------------
+    
+    // ispisuje registracije svih automobila iz Novog Sada
+    public void ispisiNS() {
+    	System.out.println("Svi automobili iz Novog Sada: ");
+    	sviIzNs(koren);
+    }
+    
+    private void sviIzNs(Cvor cvor) {
+    	if (cvor == null) 
+    		return;
+    	
+    	if (cvor.info.getRegistracija().contains("NS")) {
+    		System.out.println(cvor.info.getRegistracija());
+    	}
+    	
+    	sviIzNs(cvor.levo);
+    	sviIzNs(cvor.desno);    	
+    }
+    
+    // vraca sumu kilometraza svih plavih automobila
+    public long presliPlavi() {
+    	Suma suma = new Suma();
+    	presliPlavi(koren, suma);
+    	System.out.println("Plavi automobili su presli: ");
+    	return suma.getSuma();
+    }
+    
+    private void presliPlavi(Cvor cvor, Suma suma) {
+    	if (cvor == null)
+    		return;
+    	
+    	if (cvor.info.getBoja().equals("Plava")) {
+    		suma.setSuma(cvor.info.getKilometraza());
+    	}
+    	
+    	presliPlavi(cvor.levo, suma);
+    	presliPlavi(cvor.desno, suma);
+    }
+    
+
+    /*
+     * Pronalazi u stablu auto sa prosledjenom registracijom i
+     * kreira novo stablo u kojem pronadjeni auto ima za @param long km
+     * povecanu predjenu kilometrazu
+     */
+    public Stablo presaoJos(String registracija, long km) {
+    	Cvor korenNovog = kopijaCvora(koren, registracija, km);
+    	return new Stablo(korenNovog);
+    }
+    
+    private Cvor kopijaCvora(Cvor cvor, String reg, long km) {
+    	if (cvor == null) 
+    		return null;
+    	
+    	Cvor novi;
+    	if (cvor.info.getRegistracija().equals(reg)) {
+    		Automobil auto = new Automobil(cvor.info.getModel(), 
+    										cvor.info.getBoja(), 
+    											reg, (cvor.info.getKilometraza() + km));
+    		novi = new Cvor(auto, null, null);
+    	} else {
+    		novi = new Cvor(cvor.info, null, null);

```

Comment:
Ovo bi moglo se malo preformuliše sve pa da bude kraće za par linija pa ako oćeš možeš probati ali će ovo svakako raditi posao. Kontao sam da auto bude ono našta ga setuješ ako je registracija ista a `cvor.info` ako nije ista i to može preko jednolinijskog ifa `boolean ? true : false`. Posle toga uradiš samo novi čvor ovako:

```java
new Cvor(
	auto,
	kopijaCvora(cvor.levo, reg, km),
	kopijaCvora(cvor.desno, reg, km)
);
```

Ko što rekoh možep slobodno ostaviti i ovako, radi istu stvar na kraju samo mi se čini malo urednije ovako jer prvo gledaš koji podatak ćeš skladištiti i onda praviš kopiju.